synapseclient/models/mixins/json_schema.py

```diff
@@ -0,0 +1,1179 @@
+
+    causing_exceptions: List[CausingException]
+    """A list of causing exceptions."""
+
+
+@dataclass
+class InvalidJSONSchemaValidation:
+    """Represents the response for invalid JSON schema validation results."""
+
+    validation_response: JSONSchemaValidation
+    """The validation response object."""
+
+    validation_error_message: str
+    """A message describing the validation error."""
+
+    all_validation_messages: List[str]
+    """A list of all validation messages."""
+
+    validation_exception: ValidationException
+    """The validation exception details."""
+
+
+@dataclass
+class JSONSchemaDerivedKeys:
+    """Represents the response for derived JSON schema keys."""
+
+    keys: List[str]
+    """A list of derived keys for the entity."""
+
+
+@async_to_sync
+class BaseJSONSchema(BaseJSONSchemaProtocol):
+    """
+    Mixin class to provide JSON schema functionality.
+    This class is intended to be used with classes that represent Synapse entities.
+    It provides methods to bind, delete, and validate JSON schemas associated with the entity.
+    """
+
+    id: Optional[str] = None
+
+    async def bind_schema_async(
+        self,
+        json_schema_uri: str,
+        *,
+        enable_derived_annos: bool = False,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> JSONSchemaBinding:
+        """
+        Bind a JSON schema to the entity.
+
+        Arguments:
+            json_schema_uri: The URI of the JSON schema to bind to the entity.
+            enable_derived_annos: If true, enable derived annotations. Defaults to False.
+            synapse_client: The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            An object containing details about the JSON schema binding.
+
+        Example: Using this function
+            Binding JSON schema to a folder or a file
+
+            ```python
+                import synapseclient
+                from synapseclient.models import File, Folder
+                import asyncio
+
+                syn = synapseclient.Synapse()
+                syn.login()
+
+                # Define Project and JSON schema info
+                PROJECT_ID = syn.findEntityId(name="test_json_schema_project") #replace with your project name
+                ORG_NAME = "UniqueOrg" # replace with your organization name
+                SCHEMA_NAME = "myTestSchema" # replace with your schema name
+                VERSION = "0.0.1"
+                SCHEMA_URI = f"{ORG_NAME}-{SCHEMA_NAME}-{VERSION}"
+
+                # Create organization (if not already created)
+                js = syn.service("json_schema")
+                all_orgs = js.list_organizations()
+                for org in all_orgs:
+                    if org["name"] == ORG_NAME:
+                        print(f"Organization {ORG_NAME} already exists.")
```

Comment:
On these examples it might more sense to push folks to use an info level log messages using the Synapse Client logger instead of print statement.